### ISSUE TITLE
FR: add option "only install" to install own composer.lock

### ISSUE THREAD

**Issue by zonky2** [contao-community-alliance/composer-client#299]
add option "composer install" without "composer update" before - so its possible to install the packages without SSH-Shell if you have a own composer.lock (generated e.g. local or with https://composer.borreli.com/)

this is a workaroud if you have not enough memory...

**Comment by fritzmg** [comment]
Yes, this is a much needed feature.

**Comment by zonky2** [comment]
add a checkbox "only install" and pass the parameter update should not a rocket science...

**Comment by discordier** [comment]
> add a checkbox "only install" and pass the parameter update should not a rocket science...

Indeed, however, your PR where you implemented it has not yet appeared on my radar.

**Comment by backbone87** [comment]
Just out of curiosity: what happens if your composer.lock is not in line with your composer.json and you issue an install? (e.g. minimum-stability is set to stable, but the .lock contains dev packages?)

**Comment by fritzmg** [comment]
My suggestion would be to add this into the "Tools" section of the Package management, so that it won't confuse users (otherwise they might think they need to click this to install new packages/extensions...).

Also, the composer-client could may be automatically update the
```
"config": {
    "platform": {
        "php": "…"
    }
}
```
setting in the `/composer/composer.json` so that, if you need to execute the composer update locally, you will execute it with the platform requirement of your server, rather than your local PHP version. Not sure if this makes a whole lot of sense though.

@Toflar you said you could make a PR - do you want to do it or should I?

**Comment by Toflar** [comment]
Here's the PR for all of it: https://github.com/contao-community-alliance/composer-client/pull/308
There is no install button though, because it's executed automatically. Feel free to take whatever you need.

**Comment by fritzmg** [comment]
Cool :). But this is only for the cloud service, right? Imho an install button would still make sense.

**Comment by Toflar** [comment]
Yeah, but it handles the install command for all 3 modes too, so you might need this :)

**Comment by fritzmg** [comment]
Indeed. If this gets merged, I can create a PR for the install button ;).

**Comment by fritzmg** [comment]
> Just out of curiosity: what happens if your composer.lock is not in line with your composer.json and you issue an install? (e.g. minimum-stability is set to stable, but the .lock contains dev packages?)

I haven't checked your specific use-case, but `composer` will tell you
```
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them.
```
for example.

**Comment by discordier** [comment]
@fritzmg the current PR is in now way mergable (as I heard from @Toflar). Therefore it would be better if you could complete it with all aspects needed (cloud handling and "install button).

If all things are set, I'll happily merge your PR then.
